Argument version='det' in pauli 2017 atlas now works correctly

ryanhammonds opened the pull request:
Argument version='det' for Pauli 2017 atlas in documentation was not implemented as described.

ryanhammonds commented:
No problem! I haven't seen 'prob' and 'det' used outside of the original docstring for fetch_atlas_pauli_2017 in atlas.py. I pushed a commit updating to 'probabilisitic' and 'deterministic' for clarity.

kchawla-pi commented:
@ryanhammonds Add an entry in Whats_new under "Fixes"

ryanhammonds commented:
@kchawla-pi Added this to "whats_new".

kchawla-pi commented:
While I very much prefer the terms 'probabilistic' and 'deterministic' instead of the far less self-explanatory 'prob' & 'det', we can't drop support for them outright. While we will only document the former names, we should continue to support the latter for some time.
I will add a code review comment to that effect.

kchawla-pi commented on nilearn/datasets/atlas.py:
```diff
@@ -1228,12 +1229,12 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
     """
 
-    if version == 'prob':
+    if version == 'probabilistic':
```
To preserve backward compatibility, make it:
```python
if version == 'probabilistic' or version=='prob':
```

kchawla-pi commented on nilearn/datasets/atlas.py:
```diff
@@ -1228,12 +1229,12 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
-        filename = 'pauli_2017_labels.nii.gz'
-    elif version == 'labels':
+        filename = 'pauli_2017_probabilistic.nii.gz'
+    elif version == 'deterministic':
```
Similarly, to preserve backward compatibility, make it:
```python
if version == 'deterministic' or version=='det':
```
Do not add the prob or det to the documentation.

kchawla-pi commented on doc/whats_new.rst:
```diff
@@ -24,6 +24,7 @@ Fixes
 - `pip install nilearn` now installs the necessary dependencies.
 - :func:`nilearn.image.new_img_like` no longer attempts to copy non-iterable headers. (PR #2212)
 - Nilearn no longer raises ImportError for nose when Matplotlib is not installed.
+- Version argument corrected for fetching Pauli et al. 2017 atlas.
```
Make this more detailed:
```rst
`version` arg for :func:`nilearn.datasets.fetch_atlas_pauli_2017` now accepts the arguments "probabilistic" & "deterministic" and works as expected.
```

ryanhammonds replied:
Updated.

ryanhammonds commented:
> While I very much prefer the terms 'probabilistic' and 'deterministic' instead of the far less self-explanatory 'prob' & 'det', we can't drop support for them outright.
> 
> Much as it pains me to say this @ryanhammonds please revert the names to "prob" & "det".
> Doing anything more is beyond the purview of this PR.

 Sorry, didn't see your edit. So you would like to drop 'probabilistic' and 'deterministic' completely?

kchawla-pi commented:
> drop 'probabilistic' and 'deterministic' completely?

For this PR, yes.
Going beyond the scope of a PR makes it harder to track changes & understand history when something goes wrong or bug is introduced.

jeromedockes commented:
Note that with the code I see on master, 'det' would never have worked -- despite what the docstring says the fetcher seems to accept 'prob' or 'labels', and mix up the arguments and the corresponding filenames.

kchawla-pi commented:
> det' would never have worked 

That's what this PR was trying to fix.

ryanhammonds commented:
@kchawla-pi I reverted back to 'prob' and 'det' and updated whats_new. Sorry if I didn't address your change request properly by force pushing the changes - I'm a little new to github.

kchawla-pi commented:
HI @ryanhammonds Not at all, we all start out as new to everything. 
I think you have done the correct changes, apologies for the mixup.
I would recommend avoiding force-pushing as a general rule.

kchawla-pi commented:
Once you fix the current tests, add one which raises the `NotImplementedError`. 
You can test it with pytest thusly (import pytest in the file):
```python
with pytest.raises(NotImplementedError):
    atlas.fetch_atlas_pauli_2017('junk for testing', data_dir)
```

kchawla-pi commented on nilearn/datasets/atlas.py:
```diff
@@ -1200,8 +1200,17 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
+        Which version of the atlas should be download. This can be
+        'prob' for the probabilistic atlas or 'det' for the
+        deterministic atlas.
+>>>>>>> df260d89ffe50e15e8b63441d52667dc14f582ba
```
Delete these lines.

kchawla-pi commented on nilearn/datasets/atlas.py:
```diff
@@ -1200,8 +1200,17 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
     ----------
 
     version: str, optional (default='prob')
+<<<<<<< HEAD
```
Delete this line.

kchawla-pi commented:
HI @ryanhammonds there were some mistakes made in the last commit. I have highlighted the lines and what to do to fix them.

kchawla-pi commented on nilearn/datasets/atlas.py:
```diff
@@ -1200,8 +1200,17 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
+        Which version of the atlas should be download. This can be
+        'prob' for the probabilistic atlas or 'det' for the
+        deterministic atlas.
+>>>>>>> df260d89ffe50e15e8b63441d52667dc14f582ba
```
Delete these lines.

kchawla-pi commented on nilearn/datasets/atlas.py:
```diff
@@ -1200,8 +1200,17 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
+<<<<<<< HEAD
+        Which version of the atlas should be download. This can be 'prob' for
+        the probabilistic atlas or 'det' for the deterministic atlas.
+||||||| merged common ancestors
```
Delete these lines.

kchawla-pi commented on nilearn/datasets/tests/test_atlas.py:
```diff
@@ -555,13 +555,13 @@ def test_fetch_atlas_talairach(data_dir=tst.tmpdir):
     data_dir = os.path.join(tst.tmpdir, 'pauli_2017')
 
-    data = atlas.fetch_atlas_pauli_2017('labels', data_dir)
+    data = atlas.fetch_atlas_pauli_2017('deterministic', data_dir)
```
Replace `'deterministic'` with `'det'`.

kchawla-pi commented on nilearn/datasets/tests/test_atlas.py:
```diff
@@ -555,13 +555,13 @@ def test_fetch_atlas_talairach(data_dir=tst.tmpdir):
     assert_equal(len(np.unique(values)), 17)
 
-    data = atlas.fetch_atlas_pauli_2017('prob', data_dir)
+    data = atlas.fetch_atlas_pauli_2017('probabilistic', data_dir)
```
Replace `'probabilistic'` with `'prob'`.

kchawla-pi commented on nilearn/datasets/tests/test_atlas.py:
```diff
@@ -555,13 +555,13 @@ def test_fetch_atlas_talairach(data_dir=tst.tmpdir):
-    data = atlas.fetch_atlas_pauli_2017('prob', data_dir)
+    data = atlas.fetch_atlas_pauli_2017('probabilistic', data_dir)
     assert_equal(nibabel.load(data.maps).shape[-1], 16)
 
```
Add a test here to check the `NotImplementedError` case.

kchawla-pi commented:
HI @ryanhammonds Why the force pushes? Are you facing problems with Git? How can we help?

ryanhammonds commented:
> HI @ryanhammonds Why the force pushes? Are you facing problems with Git? How can we help?

@kchawla-pi Sorry! I try to rebase to prevent a messy commit history, and then I can't push due to conflicts. I can't figure out how to resolve the conflicts after rebasing.

kchawla-pi commented:
Simple, we don't rebase. :-)

Instead use this to pull from master
`git pull --log upstream master`

Sometimes git can't resolve the merge and conflicts happen. That's fine. Examine each conflict carefully and edit it. 

This is easier with a GUI based Git Editor. Pycharm has one built in, I can help you with that if you use pycharm.

If you use some other editor, tell me and we can find a solution.

kchawla-pi commented:
Nilearn does not endorse rebasing. It would be helpful for us if you don't rebase for NIlearn PRs.

ryanhammonds commented:
Thanks! I won't rebase in the future, apologies. I'll need to figure out how to resolve conflicts more efficiently. How does Nilearn prefer to clean up commit history, for example fixing a typo that could be included a previous commit? I typically use Atom but also have Pycharm installed.

kchawla-pi commented:
We don't touch the commit history of PRs. A standard merge is what we go for. When the PR is ready we Squash merge it. This keeps the master's commit history easily parsable and we can go ot the individual PRs for the detailed history.

kchawla-pi commented:
Force pushes can cause problems and more work for the author.
I would recommend getting a good plugin for your editor to handle git merges & conflicts better.
If you don't find any or are still confused, ping me and I'll help you out.

ryanhammonds commented:
I see, makes sense. Sorry for the mess of a very simple pull request. I will keep this in mind for all future PR. Thanks for your patience!

kchawla-pi commented:
@ryanhammonds there is nothing to apologize for, all of us at Nilearn have been here many times, and will be here in future as well, messing up our commit histories.. It's a process. I was terrified that I will break the world with one wrong git merge when I started.
Thank you so much for helping us out. Do not hold back from submitting PRs in the future.
